h2o-algos/src/main/java/hex/rulefit/RuleEnsemble.java

```diff
@@ -75,23 +70,48 @@ else if (filteredRule.size() > 1) {
     }
 }
 
+class Transform extends MRTask<Transform> {
+    Rule[] _rules;
+    String[] _names;
+
+    Transform(Rule[] rules, String[] names) {
+        _rules = rules;
+        _names = names;
+    }
+
+    @Override
+    public void map(Chunk[] cs, NewChunk[] ncs) {
+        byte[][] bytes = new byte[_rules.length][];
```

Comment:
you actually only need one bytearray, should be argument sent to rule.transform()

Reply:
fixed